src/composition/supergraph/binary.rs

```diff
@@ -120,6 +122,62 @@ impl SupergraphBinary {
                 error: format!("{err:?}"),
             })
     }
+
+    pub async fn run_connector(
+        &self,
+        exec_impl: &impl ExecCommand,
+        schema_path: PathBuf,
+        connector_id: String,
+        variables: String,
+    ) -> Result<RoverOutput, RunConnectorError> {
+        let args = vec![
+            "run-connector".to_string(),
+            "--path".to_string(),
+            schema_path.to_str().unwrap_or_default().into(),
+            "--connector-id".to_string(),
+            connector_id,
+            "--variables".to_string(),
+            variables,
+        ];
+
+        let config = ExecCommandConfig::builder()
+            .exe(self.exe.clone())
+            .args(args)
+            .output(ExecCommandOutput::builder().stdout(Stdio::piped()).build())
+            .build();
+
+        let output = exec_impl
+            .exec_command(config)
+            .await
+            .tap_err(|err| tracing::error!("{:?}", err))
+            .map_err(|err| RunConnectorError::Binary {
+                error: format!("{err:?}"),
+            })?;
+
+        let exit_code = output.status.code();
+        if exit_code != Some(0) && exit_code != Some(1) {
+            return Err(RunConnectorError::BinaryExit {
+                exit_code,
+                stdout: String::from_utf8(output.stdout).unwrap(),
+                stderr: String::from_utf8(output.stderr).unwrap(),
```

Comment:
```suggestion
                stdout: String::from_utf8(output.stdout),
                stderr: String::from_utf8(output.stderr),
```

No panic 😓. Maybe `.unwrap_or_default()` if those values are required?

Reply:
This is a copy/paste of what the existing `compose` command does 😅 

With that in mind, do you still want me to change this?